src/common/users.ts

```diff
@@ -431,6 +444,32 @@ export const shouldResetStreak = (
   return day > firstDayOfWeek && difference > MISSED_LIMIT;
 };
 
+export const checkRestoreValidity = (
+  day: number,
+  difference: number,
+  startOfWeek: DayOfWeek = DEFAULT_WEEK_START,
+) => {
+  const firstDayOfWeek =
+    startOfWeek === DayOfWeek.Monday ? Day.Monday : Day.Sunday;
+
+  const lastDayOfWeek =
+    startOfWeek === DayOfWeek.Monday ? Day.Sunday : Day.Saturday;
+
+  const getAllowedDays = () => {
+    if (day === lastDayOfWeek) {
+      return FREEZE_DAYS_IN_A_WEEK;
+    }
+
+    if (day === firstDayOfWeek) {
+      return FREEZE_DAYS_IN_A_WEEK + STREAK_RECOVERY_MAX_GAP_DAYS;
+    }
+
+    return STREAK_RECOVERY_MAX_GAP_DAYS;
+  };
```

Comment:
lets move function to outside so its not inline initiated each time

Reply:
Agreed. Should be moved by now.